source/BayType.h

```diff
@@ -0,0 +1,54 @@
+/* BayType.h
+Copyright (c) 2024 by Amazinite
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef BAY_TYPE_H_
+#define BAY_TYPE_H_
+
+#include <set>
+#include <string>
+
+class DataNode;
+
+
+
+// A BayType is a list of ship categories that can be stored in a
+// bay point on a Ship that shares the name of this BayType.
+class BayType {
+public:
+	BayType() = default;
+	// Construct and Load() at the same time.
+	BayType(const DataNode &node);
+
+	void Load(const DataNode &node);
+	// Confirm that all categories of ship that this bay can hold
+	// are carriable. If not, set isValid to false.
+	void FinishLoading();
+
+	bool Contains(const std::string &category) const;
+	const std::set<std::string> &Categories() const;
+
+	bool IsLoaded() const;
+	bool IsValid() const;
+
+
+private:
+	std::string name;
+	std::set<std::string> categories;
```

Comment:
This is just a side note, but I did some reading on the stl recently and it seems like a std set is actually slower than a std vector in this situation, especially with such a low element count.

To give an example, this code:
```c++
#include <set>
#include <string>
#include <vector>
int main()
{
    std::vector<std::string> testE = {"a", "b", "c", "d", "e"};
    std::set<std::string> test = {"a", "b", "c", "d"};
    for(int x = 0; x < 200000000; x++)
        int i = test.count(testE[x % 3]);
    return 0;
}
```

has a worse runtime than this one:
```c++
#include <vector>
#include <string>
#include <algorithm>
int main()
{
    std::vector<std::string> testE = {"a", "b", "c", "d", "e"};
    std::vector<std::string> test = {"a", "b", "c", "d"};
    for(int x = 0; x < 200000000; x++)
        bool it = std::find(test.begin(), test.end(), testE[x % 3]) == test.end();
    return 0;
}
```
This does not really matter, but I just wanted to note this here as I saw a lot of those really useless set use-cases.

Reply:
Eh. `set.count(val)` is more readable than `find(vector.begin(), vector.end(), val) != vector.end()`. Even if the runtime is worse, it doesn't actually result in any tangible difference at the scales we're working at. Checking the categories of a bay isn't exactly a common action, relatively speaking. So this is one of those "We'll change it if it ever actually causes a problem" things.

Reply:
yea was just giving random thoughts